### PR TITLE
M3-3640 Accessibility: Focus on content when navigating

### PR DESCRIPTION
**Pull request by alioso**
## Accessibility: Focus on content when navigating

This PR improves accessibility by adding focus on the H1 when navigating to a new page. Since this is a one page app, our router does not do that automatically, so I implemented the focus() function in the breadcrumb (and editable text component since it also contains the H1) and replaced the breadcrumb wherever an isolated `<Typography variant="h1" />` was found. This should be the pattern to follow for ALL page titles from now on to preserve this enhancement.

## Type of Change
- Non breaking change ('update', 'change')

## Note to Reviewers

1. Please review EditableText component as i refactored it as a functional component with hooks
2. You'll notice the css for focus is a red outline. This is intentional for everyone to test, and will be removed once PR is approved

### CODE REVIEW

File: packages/manager/src/components/EditableText/EditableText.tsx [hunk 1]
```diff
@@ -149,168 +149,156 @@ interface Props {
...
 
 type FinalProps = PassThroughProps & WithStyles<ClassNames>;
 
-export class EditableText extends React.Component<FinalProps, State> {
-  state: State = {
-    isEditing: Boolean(this.props.errorText),
-    text: this.props.text
-  };
+const EditableText: React.FC<FinalProps> = props => {
+  const [isEditing, setisEditing] = React.useState(Boolean(props.errorText));
```
Comment (johnwcallahan) [inline packages/manager/src/components/EditableText/EditableText.tsx]: Minor: casing of `setIsEditing`.

**Comment by alioso** [comment]
@johnwcallahan good calls all around. Refactored to use a new component (unless where breadcrumb made sense)

File: packages/manager/src/components/EditableText/EditableText.tsx [hunk 2]
```diff
@@ -149,168 +149,148 @@ interface Props {
...
 
-export class EditableText extends React.Component<FinalProps, State> {
-  state: State = {
-    isEditing: Boolean(this.props.errorText),
-    text: this.props.text
-  };
+const EditableText: React.FC<FinalProps> = props => {
+  const [isEditing, setIsEditing] = React.useState(Boolean(props.errorText));
+  const [text, setText] = React.useState(props.text);
+  const h1Header = React.useRef<HTMLDivElement>(null);
```
Comment (johnwcallahan) [inline packages/manager/src/components/EditableText/EditableText.tsx]: I think you can remove this now, since this is built-in to the H1Header component.

File: packages/manager/src/components/H1Header/H1Header.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+interface Props {
+  title: string;
+  className?: string;
+}
+
+const H1Header: React.FC<Props> = props => {
+  const h1Header = React.useRef<HTMLDivElement>(null);
+  const { className, title } = props;
+
+  React.useEffect(() => {
```
Comment (johnwcallahan) [inline packages/manager/src/components/H1Header/H1Header.tsx]: A quick comment explaining the reason for this logic (accessibility issues) may be helpful in the future.

File: packages/manager/src/features/linodes/LinodesCreate/LinodeCreateContainer.tsx [hunk 4]
```diff
@@ -565,9 +565,11 @@ class LinodeCreateContainer extends React.PureComponent<CombinedProps, State> {
         <DocumentTitleSegment segment="Create a Linode" />
         <Grid container spacing={0}>
           <Grid item xs={12}>
-            <Typography variant="h1" data-qa-create-linode-header>
-              Create New Linode
-            </Typography>
+            <Breadcrumb
+              pathname={this.props.location.pathname}
+              labelTitle="Create"
```
Comment (johnwcallahan) [inline packages/manager/src/features/linodes/LinodesCreate/LinodeCreateContainer.tsx]: Just checking... is this change intentional? 

Screen Shot 2019-12-04 at 3 16 53 PM

I don't love that you can create UI elements by adding junk to the URL... could we maybe hardcode the `pathname` here? ("/linodes/create")

File: packages/manager/src/components/EditableText/EditableText.tsx [hunk 5]
```diff
@@ -149,168 +149,141 @@ interface Props {
...
   };
 
-  cancelEditing = () => {
-    /** cancel editing and invoke callback function and revert text to original */
-    this.setState({ isEditing: false, text: this.props.text }, () => {
-      this.props.onCancel();
-    });
+  const cancelEditing = () => {
+    setIsEditing(false);
+    setText(props.text);
```
Comment (martinmckenna) [inline packages/manager/src/components/EditableText/EditableText.tsx]: this isn't calling `props.onCancel` anymore

File: packages/manager/src/components/H1Header/H1Header.tsx [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+      h1Header.current.focus();
+    }
+  }, []);
+
+  return (
+    <Typography variant="h1" className={className} ref={h1Header} tabIndex={-1}>
+      {title}
+    </Typography>
+  );
+};
```
Comment (martinmckenna) [inline packages/manager/src/components/H1Header/H1Header.tsx]: I'm very down with this, but just for future reference, this is the exact reason we abstracted out the MUI components.

What we could have done is made edits to `src/components/Typography` directly and had something like

```js

  React.useEffect(() => {
    if (props.variant === 'h1' && h1Header.current !== null) {
      h1Header.current.focus();
    }
  }, []);
```

which in turn could have made this a much smaller PR

File: packages/manager/src/components/EditableText/EditableText.tsx (hunk 5 above)
Comment (alioso) [reply packages/manager/src/components/EditableText/EditableText.tsx]: fix in https://github.com/linode/manager/pull/5858

File: packages/manager/src/components/H1Header/H1Header.tsx (hunk 6 above)
Comment (alioso) [reply packages/manager/src/components/H1Header/H1Header.tsx]: Right, except we don't have a Typography component, so we'd have then to create one and  change many imports...
